app/assets/javascripts/Components/__tests__/submission_file_manager.test.jsx

```diff
@@ -0,0 +1,90 @@
+import {SubmissionFileManager} from "../submission_file_manager";
+
+import {mount} from "enzyme";
+
+describe("For the submissions managed by SubmissionFileManager's FileManager child component", () => {
+  let wrapper, rows;
+  const files_sample = [
+    {
+      id: 136680,
+      url: "test.url",
+      filename: '<img src="" /><a href=""> HelloWorld.java</a>',
+      raw_name: "HelloWorld.java",
+      last_revised_date: "Saturday, May 14, 2022, 09:15:24 PM EDT",
+      last_modified_revision: "58ca2e15254aa63c4d41cb5db7dfc398b6bda3fb",
+      revision_by: "c5anthei",
+      submitted_date: "Saturday, May 14, 2022, 09:15:24 PM EDT",
+      type: "java",
+      key: "HelloWorld.java",
+      modified: 1652577324,
+      relativeKey: "HelloWorld.java",
+    },
+    {
+      id: 136700,
+      url: "test2.url",
+      filename: '<img src="" /><a href=""> deferred-process.java</a>',
+      raw_name: "deferred-process.java",
+      last_revised_date: "Saturday, May 14, 2022, 09:15:24 PM EDT",
+      last_modified_revision: "58ca2e15254aa63c4d41cb5db7dfc398b6bda3fb",
+      revision_by: "c5anthei",
+      submitted_date: "Saturday, May 14, 2022, 09:15:24 PM EDT",
+      type: "java",
+      key: "deferred-process.java",
+      modified: 1652577324,
+      relativeKey: "deferred-process.java",
+    },
+  ];
+
+  beforeEach(() => {
+    // Unlike FileManager, files are stored in SubmissionFileManager's states, which are set when the component mounts
+    // and calls fetchData. As a result, we need to mock that fetch to return our data.
+    // We need to mock "twice" (i.e. two promises) because of how fetch works.
```

Comment:
Okay I looked into this a bit, and even though what you have is find for this case, I'd rather use a library like https://github.com/jefflau/jest-fetch-mock (which is well-maintained) to do this mocking. Please introduce that in this PR as well, so that we have a good example of its use for the future. :)